include/mega/sync.h

```diff
@@ -289,20 +599,39 @@ struct Syncs
 
     unique_ptr<BackupMonitor> mHeartBeatMonitor;
 
-    // use this existing class for maintaining the db
-    unique_ptr<SyncConfigBag> mSyncConfigDb;
+    // Returns a reference to this user's internal configuration database.
+    JSONSyncConfigDB* syncConfigDB();
+
+    // Whether the internal database has changes that need to be written to disk.
+    bool syncConfigDBDirty();
+
+    // Attempts to flush the internal configuration database to disk.
+    error syncConfigDBFlush();
+
+    // Load internal sync configs from disk.
+    error syncConfigDBLoad();
 
 private:
+    // Returns a reference to this user's sync config IO context.
+    JSONSyncConfigIOContext* syncConfigIOContext();
```

Comment:
Access to the context goes through this method so that the context can be lazily created.

This is currently necessary as we need the values of the ^~jscn and ^~jsck user attributes in order to create the context.